cpu/cortexm_common/thread_arch.c

```diff
@@ -337,8 +345,14 @@ void __attribute__((naked)) __attribute__((used)) isr_pendsv(void) {
 
     /* current thread context is now saved */
     "restore_context:                 \n" /* Label to skip thread state saving */
-
-    "ldr    r0, [r0]                  \n" /* load tcb->sp to register 1 */
+#ifdef MODULE_CORTEXM_STACK_LIMIT
+    "mov    r2, r0                    \n" /* Save content of R0 into R2*/
+    "bl _get_new_stacksize            \n" /* Get the new lower limit stack in R0 */
```

Comment:
It has been quite some time since I've last have written thumb asm, so maybe I'm just a bit lost here. But if I recall correctly r0 to r3 are used to pass arguments and return values and are assumed to be saved by the caller. In that case, `r2` could be overwritten by the code `_get_new_stacksize()` emits, right?

Reply:
You're right.
When I looked at objdump, only R0 and R3 were used by this function so I just kept R2 to temporary save R0 content.
I'll try to use another register, it shouldn't be too much pain.

Reply:
I've moved to R4 for storing R0 content now. This register is restored a few instructions after anyways so it shouldn't be an issue.